packages/mdc-slider/foundation.js

```diff
@@ -0,0 +1,420 @@
+      setTrackStyleProperty: (/* propertyName: string, value: string */) => {},
+      isRTL: () => /* boolean */ false,
+    };
+  }
+
+  constructor(adapter = {}) {
+    super(Object.assign(MDCSliderFoundation.defaultAdapter, adapter));
+    this.rect_ = null;
+    // We set this to NaN since we want it to be a number, but we can't use '0' or '-1'
+    // because those could be valid tabindices set by the client code.
+    this.savedTabIndex_ = NaN;
+    this.off_ = false;
+    this.active_ = false;
+    this.inTransit_ = false;
+    this.handlingThumbTargetEvt_ = false;
+    this.min_ = 0;
+    this.max_ = 100;
+    this.step_ = 0;
+    this.value_ = 0;
+    this.disabled_ = false;
+    this.preventFocusState_ = false;
+    this.updateUIFrame_ = 0;
+    this.thumbContainerPointerHandler_ = () => {
+      this.handlingThumbTargetEvt_ = true;
+    };
+    this.mousedownHandler_ = this.createDownHandler_('mousemove', 'mouseup');
+    this.pointerdownHandler_ = this.createDownHandler_('pointermove', 'pointerup');
+    this.touchstartHandler_ = this.createDownHandler_(
+      'touchmove', 'touchend', ({targetTouches}) => targetTouches[0].pageX);
+    this.keydownHandler_ = (evt) => this.handleKeydown_(evt);
+    this.focusHandler_ = () => this.handleFocus_();
+    this.blurHandler_ = () => this.handleBlur_();
+    this.resizeHandler_ = () => this.layout();
+  }
+
+  init() {
+    this.adapter_.registerInteractionHandler('mousedown', this.mousedownHandler_);
+    this.adapter_.registerInteractionHandler('pointerdown', this.pointerdownHandler_);
+    this.adapter_.registerInteractionHandler('touchstart', this.touchstartHandler_);
+    this.adapter_.registerInteractionHandler('keydown', this.keydownHandler_);
+    this.adapter_.registerInteractionHandler('focus', this.focusHandler_);
+    this.adapter_.registerInteractionHandler('blur', this.blurHandler_);
+    ['mousedown', 'pointerdown', 'touchstart'].forEach((evtName) => {
+      this.adapter_.registerThumbContainerInteractionHandler(evtName, this.thumbContainerPointerHandler_);
+    });
+    this.adapter_.registerResizeHandler(this.resizeHandler_);
+    this.layout();
+  }
+
+  destroy() {
+    this.adapter_.deregisterInteractionHandler('mousedown', this.mousedownHandler_);
+    this.adapter_.deregisterInteractionHandler('pointerdown', this.mousedownHandler_);
+    this.adapter_.deregisterInteractionHandler('touchstart', this.mousedownHandler_);
+    this.adapter_.deregisterInteractionHandler('keydown', this.keydownHandler_);
+    this.adapter_.deregisterInteractionHandler('focus', this.focusHandler_);
+    this.adapter_.deregisterInteractionHandler('blur', this.blurHandler_);
+    ['mousedown', 'pointerdown', 'touchstart'].forEach((evtName) => {
+      this.adapter_.deregisterThumbContainerInteractionHandler(evtName, this.thumbContainerPointerHandler_);
+    });
+    this.adapter_.deregisterResizeHandler(this.resizeHandler_);
+  }
+
+  layout() {
```

Comment:
I think that's fine. When you update an `HTMLInputElement`'s `value` prop programmatically you don't get an event either, so this keeps our behavior consistent with the way the DOM API behaves.